src/parser.ts

```diff
@@ -3780,14 +3786,20 @@ export class Parser extends DiagnosticEmitter {
 
                 // if we got here, check for arrow
                 case Token.CloseParen: {
-                  if (
-                    !tn.skip(Token.Colon) &&
-                    !tn.skip(Token.Equals_GreaterThan)
-                  ) {
-                    again = false;
-                    break;
+                  // `Identifier):Type =>` is function expression
+                  if (tn.skip(Token.Colon)) {
+                    let type = this.parseType(tn, true, true);
+                    if (type == null) {
+                      again = false;
+                      break;
+                    }
                   }
-                  // fall-through
+                  if (tn.skip(Token.Equals_GreaterThan)) {
+                    tn.reset(state);
+                    return this.parseFunctionExpression(tn);
+                  }
+                  again = false;
+                  break;
```

Comment:
Seems that when inverting the condition, the fall-through to parsing a function expression can be maintained instead of duplicating it?
```suggestion
                  if (!tn.skip(Token.Equals_GreaterThan)) {
                    again = false;
                    break;
                  }
                  // fall-through
```